xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/register_macros.vm

```diff
@@ -0,0 +1,217 @@
+  #set ($discard = $xwiki.jsfx.use('uicomponents/widgets/validation/livevalidation_prototype.js'))
+  <script type='text/javascript'>
+  /* <![CDATA[ */
+  var initRegistrationFormValidation = function() {
+  ##
+  #foreach ($field in $fields)
+    #if ($field.validate && $field.name)
+      #set ($validate = $field.validate)
+      #if (($validate.mandatory && !$validate.mandatory.noscript)
+          || ($validate.regex && !$validate.regex.noscript)
+          || ($validate.mustMatch) && !$validate.mustMatch.noscript)
+        #set ($fieldName = $field.name)
+        #if ($validate.fieldOkayMessage)
+          #set ($okayMessage = $validate.fieldOkayMessage)
+        #elseif (!$validate.hideOkayMessage)
+          #set ($okayMessage = $services.localization.render('core.validation.valid.message'))
+          #else
+            #set ($okayMessage = '')
+        #end
+        var ${fieldName}Validator = new LiveValidation("$fieldName", {validMessage: "$okayMessage", wait: 500});
+        ##
+        #if ($validate.mandatory)
+          #set ($mandatory = $validate.mandatory)
+          #if ($mandatory.failureMessage && !$mandatory.noscript)
+            ${fieldName}Validator.add(Validate.Presence, {failureMessage: "$!mandatory.failureMessage"});
+          #end
+        #end
+        ##
+        #if ($validate.mustMatch)
+          #set ($mustMatch = $validate.mustMatch)
+          #if ($mustMatch.name && $mustMatch.failureMessage && !$mustMatch.noscript)
+            ${fieldName}Validator.add(Validate.Confirmation, {match: $$("input[name=$!mustMatch.name]")[0], 
+              failureMessage: "$!mustMatch.failureMessage"});
+          #end
+        #end
+        ##
+        #if ($validate.regex)
+          #set ($regex = $validate.regex)
+          #set ($pattern = "")
+          #if ($regex.jsPattern)
+            #set ($pattern = $regex.jsPattern)
+          #elseif ($regex.pattern)
+            #set ($pattern = $regex.pattern)
+          #end
+          #set ($failMessage = "")
+          #if ($regex.jsFailureMessage)
+            #set ($failMessage = $regex.jsFailureMessage)
+          #elseif ($regex.failureMessage)
+            #set ($failMessage = $regex.failureMessage)
+          #end
+          #if ($pattern != '' && $failMessage != '' && !$regex.noscript)
+            ${fieldName}Validator.add(Validate.Format, {pattern: $pattern, failureMessage: "$failMessage"});
+          #end
+        #end
+      #end
+    #end
+  #end
+    };
+    document.observe('xwiki:dom:loaded', initRegistrationFormValidation);
+    document.observe('xwiki:dom:updated', function(event) {
+      var container = (event && event.memo && event.memo.elements && event.memo.elements[0]) || $('body');
+      if (container.down('form#register')) {
+        initRegistrationFormValidation();
+      }
+    });// ]]>
+    </script>
+#end
+##
+#macro (outputPasswordFields, $fields, $passwordFieldName, $confirmPasswordFieldName, $request)
```

Comment:
Where is this macro actually used? I cannot find it in the diff or in the project?

Reply:
As the way it was designed first time, it was supposed to be used for change password form. After that, changes were needed and it's not used any more. FTM, it's unused.

Thanks for catching that.

Reply:
OK I'll remove it on my next commit then. I'm still doing some changes regarding the validation.